app/controllers/api/v1/sessions_controller.rb

```diff
@@ -41,20 +41,27 @@ def session_params
       end
 
       def sign_in(user)
+        user.generate_session_token!(extended_session: session_params[:extend_session])
+
         # Creates an extended_session cookie if extend_session is selected in sign in form.
         if session_params[:extend_session]
           cookies.encrypted[:_extended_session] = {
             value: {
-              user_id: user.id
+              session_token: user.session_token,
+              session_expiry: user.session_expiry
             },
             expires: 7.days,
```

Comment:
I think it would be better to extract the expiration periods into a centralized entity like a constant.
The expiration period looks to be used here and in the `User#generate_session_token!` which introduces if not carefully synchronized a good room for error,
However, that does not seem to have a security concern.